code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/dao/impl/ServiceExposeMapDaoImpl.java

```diff
@@ -92,24 +55,10 @@ public void run() {
                         .and(SERVICE_EXPOSE_MAP.SERVICE_ID.eq(serviceId))
                         .and(SERVICE_EXPOSE_MAP.STATE.in(CommonStatesConstants.ACTIVATING,
                                 CommonStatesConstants.ACTIVE, CommonStatesConstants.REQUESTED)))
```

Comment:
for serviceExposeMap, I don't kick off the activation process till the vm starts. Let me explain why. ServiceExposeMap activation triggers DNS and healthcheck updates, and for those config items the IP address has to be set on the instance already. So I couldn't think of better handling for it then:

* serviceExposeMap gets created along with the instanceCreate.
* Then instance.start triggers serviceExposeMap.activate
* serviceExposeMap.activate triggers DNS and Healthcheck update

I'm open for better solution suggestion :)